### PR TITLE
updated to keras spec

### PR DESCRIPTION
**Pull request by DeclanMcIntosh**
updated to Keras spec as seen in https://stanford.edu/~shervine/blog/keras-how-to-generate-data-on-the-fly

Is one monolithic commit because I'm a bad person...

### CODE REVIEW

**Comment by DeclanMcIntosh** [comment]
Can you check the interface of the data I provide is OK for your loss function
